objects.py

```diff
@@ -43,13 +97,46 @@ def __init__(self, *args, **kwargs):
         json.JSONDecoder.__init__(self, object_hook=self.object_hook, *args, **kwargs)
 
     def object_hook(self, obj):
-        if '_type' not in obj:
-            return obj
-        type = obj['_type']
+        type = obj.get('_type', '')
         if type == 'spec_config':
             return spec_config(obj)
+        templates = obj.get('TemplateData', '')
+        if (templates != ''):
+            return self.decode_tate(obj, templates)
         return obj
 
+    def decode_tate(self, obj, templates):
+        runlist = obj.get('RunList', '')
+        if (runlist == ''):
+            raise AssertionError("Json file has no run lists")
+        #Run lists is a list, templates is a dict.
+        ret = spec_config()
+        ret.type = "tate_config"
+        for run in runlist:
+            r = spec_run()
+            template_type = run.get('template_type', '')
```

Comment:
Template type is required, so if this isn't provided you could also just throw an exception:
```python
try:
    run.get("template_type")
except IndexError:
    raise AssertionError("it's gone")
```